src/object/srv_obj_migrate.c

```diff
@@ -832,6 +833,7 @@ migrate_fetch_update_single(struct migrate_one *mrone, daos_handle_t oh,
 		D_ASSERT(mrone->mo_iods[i].iod_type == DAOS_IOD_SINGLE);
 
 		size = daos_iods_len(&mrone->mo_iods[i], 1);
+		D_ASSERT(size != -1);
 		D_ALLOC(data, size);
```

Comment:
in PR3389, as we discussed this func (migrate_fetch_update_single) need some refining -
"here how to differentiate which parity target's parity data's address?
in above obj_ec_singv_encode_buf() returned e_iov, include all parity shards' data, need to take this parity shard's part right?
for example for 4+2 EC obj, when rebuild shard 5, but e_iov.iov_buf with parity data for both shard 4 and shard 5. ."

but see below part of code, that change is not included in this patch? probably the long singv rebuild one parity shard is not covered in testing?

Reply:
ah, right.